metadata.md

```diff
@@ -0,0 +1,48 @@
+# Metadata
+At the top of every feature detect in Modernizr there is a JSON fragment that represents the metadata of the test. This data is used, for example, to build the webpage.
+This is an example of this schema (it does not represent a real test): 
+## Schema
+```json
+/*!
+{
+  "name": "JPEG 2000",
+  "property": "jpeg2000",
+  "tags": ["media", "attribute"],
+  "caniuse": "jpeg2000",
+  "authors": ["Markel Ferro (@MarkelFe)", "@rejas", "Brandom Aaron"],
+  "builderAliases": ["a_download"],
+  "polyfills": ["xaudiojs"],
+  "aliases": ["jpeg-2000", "jpg2"],
+  "async": true,
+  "warnings": ["These tests currently require document.body to be present"],
+  "knownBugs": ["This will false positive in IE6"],
+  "notes": [{
+    "name": "Specification",
+    "href": "https://www.w3.org/"
+  }, {
+    "name": "Github issue",
+    "href": "https://github.com/Modernizr/"
+  }]
+}
+!*/
+/* DOC
+Here it would go a description of the feature test. You can use **markdown** here :)
+*/
+```
+> There is no set order in which they must appear, but it is common to see `name` and `property` at the top while `notes` at the bottom.
+
+## Item description
+|                  | Necesity |                      Description                     |                                  Notes                                  |
+|------------------|:--------:|:----------------------------------------------------:|:-----------------------------------------------------------------------:|
+| `name`           | required |             Name of the feature detection            |                                                                         |
```

Comment:
[technically this is not required](https://github.com/Modernizr/Modernizr/blob/870cf370ab370223199462897abfbf0ec5469878/lib/metadata.js#L69-L71). 

nothing really is, honestly.

Reply:
@rejas do you think metadata.js should throw if these are not provided?

Reply:
I mean, I used [this file](https://github.com/Modernizr/Modernizr/blob/870cf370ab370223199462897abfbf0ec5469878/test/node/lib/metadata.js#L159) to determine which things are necessary. Although it may be completely unrelated, I don't know the entire project architecture yet.

Reply:
good find! That tests the output of the metadata, this doc is about the input. What you linked to makes sure that what I linked to if performed. 
I am fine with it being explicitly required, but I think if we do we should make sure the existing metadata passes and make lib/metadata.js throw if it is missing

Reply:
i think "required" should be the stuff that is absolutelty minium for the website to display something sensible. If it is missing then there should be a warning thrown